app/models/ability.rb

```diff
@@ -45,14 +45,20 @@ def not_signed_in
     # can view Commercials of confirmed Events
     can :show, Commercial, commercialable_type: 'Event', commercialable_id: Event.where(state: 'confirmed').pluck(:id)
     can [:show, :create], User
-    unless ENV['OSEM_ICHAIN_ENABLED'] == 'true'
+    if ENV['OSEM_ICHAIN_ENABLED'] == 'true'
+      can [:new, :create], Registration do |registration|
```

Comment:
I am missing something here. In the controller method, we're redirecting the user to ichain login page when he tries to create a new registration. So, the user can only  create a new registration after he's signed in right?

Or am I wrong?